### PR TITLE
Port adjust-times to Windows

### PR DESCRIPTION
**Pull request by gmittert**
Switch out getopt for llvm's Commandline API which is cross platform,
and define various constants which aren't normally defined on Windows.

This allows the Windows build to compile again.

### CODE REVIEW

**Comment by gmittert** [comment]
This does change the exposed options to use llvm style single `-` instead of `--`.  I'm assuming this isn't used anywhere except in the tests?

cc @vlm

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by gmittert** [comment]
Ah, I had to update the XCode build :( Thanks @compnerd for helping me figure out how to have XCode set up the link dependency.

@swift-ci please smoke test

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Comment by jakepetroules** [comment]
@gmittert Awesome, thanks for doing this!

@dmbryson @ddunbar Do we care about `-` vs `--` or does it not matter?

**Comment by vlm** [comment]
@jakepetroules we don't care about `-` vs `--`.

File: utils/adjust-times/adjust-times.cpp [hunk 1]
```diff
@@ -190,79 +201,51 @@ int main(int argc, char **argv) {
...
+      errno = ERANGE;
+    }
+    if (errno != 0) {
+      fprintf(stderr, "--adjust %f: %s\n", value, strerror(errno));
+      exit(EX_DATAERR);
+    }
+    adjust_relative_sec = Adjust;
+    mode = ModeAdjustMtime;
+  }
+  if (!File.empty()) {
```
Comment (vlm) [inline utils/adjust-times/adjust-times.cpp]: nit: `s/File/Filename/`?

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Comment by vlm** [comment]
@swift-ci please smoke test

File: utils/adjust-times/adjust-times.cpp [hunk 2]
```diff
@@ -1,33 +1,31 @@
...
-#include <libgen.h>
 #include <time.h>
-#include <sysexits.h>
 #include <sys/stat.h>
-#include <sys/time.h>
 #include <fcntl.h>
 #include <errno.h>
 #include <math.h>
 #include <string.h>
+#include "llvm/Support/Commandline.h"
```
Comment (vlm) [inline utils/adjust-times/adjust-times.cpp]: Please use `CommandLine` (case-sensitivity).

Comment (gmittert) [reply utils/adjust-times/adjust-times.cpp]: Windows!! :( Fixed now, thanks.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by vlm** [comment]
@swift-ci please smoke test

**Comment by vlm** [comment]
@swift-ci please smoke test

**Comment by vlm** [comment]
Closing in favor https://github.com/apple/swift-llbuild/pull/600 (containing a couple other fixes on top of @gmittert's changes)

**Pull request closed without merging**